src/index.ts

```diff
@@ -0,0 +1,231 @@
+import * as util from 'util';
+
+const toString = Object.prototype.toString;
+const getObjectType = (value: any) => toString.call(value).slice(8, -1) as string;
+const isOfType = (type: string) => (value: any) => typeof value === type; // tslint:disable-line
+const isObjectOfType = (type: string) => (value: any) => getObjectType(value) === type;
+
+function is(value: any) { // tslint:disable-line:only-arrow-functions
+	if (value === null) {
+		return 'null';
+	}
+
+	if (value === true || value === false) {
+		return 'boolean';
+	}
+
+	const type = typeof value;
+
+	if (type === 'undefined') {
+		return 'undefined';
+	}
+
+	if (type === 'string') {
+		return 'string';
+	}
+
+	if (type === 'number') {
+		return 'number';
+	}
+
+	if (type === 'symbol') {
+		return 'symbol';
+	}
+
+	if (is.function_(value)) {
+		return 'Function';
+	}
+
+	if (Array.isArray(value)) {
+		return 'Array';
+	}
+
+	if (Buffer.isBuffer(value)) {
+		return 'Buffer';
+	}
+
+	const tagType = getObjectType(value);
+	if (tagType) {
+		return tagType;
+	}
+
+	if (value instanceof String || value instanceof Boolean || value instanceof Number) {
+		throw new TypeError('Please don\'t use object wrappers for primitive types');
+	}
+
+	return 'Object';
+}
+
+namespace is { // tslint:disable-line:no-namespace
+	const isObject = (value: any) => typeof value === 'object';
+
+	// tslint:disable:variable-name
+	export const undefined = isOfType('undefined');
+	export const string = isOfType('string');
+	export const number = isOfType('number');
+	export const function_ = isOfType('function');
+	export const null_ = (value: any) => value === null;
+	export const class_ = (value: any) => function_(value) && value.toString().startsWith('class ');
+	export const boolean = (value: any) => value === true || value === false;
+	// tslint:enable:variable-name
+
+	export const symbol = isOfType('symbol');
+
+	export const array = Array.isArray;
+	export const buffer = Buffer.isBuffer;
+
+	export const nullOrUndefined = (value: any) => null_(value) || undefined(value);
+	export const object = (value: any) => !nullOrUndefined(value) && (function_(value) || isObject(value));
+	export const iterable = (value: any) => !nullOrUndefined(value) && function_(value[Symbol.iterator]);
+	export const generator = (value: any) => iterable(value) && function_(value.next) && function_(value.throw);
+
+	export const nativePromise = isObjectOfType('Promise');
+
+	const hasPromiseAPI = (value: any) =>
+		!null_(value) &&
+		isObject(value) &&
+		function_(value.then) &&
+		function_(value.catch);
+
+	export const promise = (value: any) => nativePromise(value) || hasPromiseAPI(value);
+
+	// TODO: Change to use `isObjectOfType` once Node.js 6 or higher is targeted
```

Comment:
Can this be used now?

Reply:
no, the comment is still valid because the restricting factor is node runtime here